### PR TITLE
Fixed Reverse Distillation export to ONNX

### LINKED ISSUE #1513: [Bug]: ONNX Export for Reverse Distillation fails
### Describe the bug

ONNX export fails with error
```bash
torch.onnx.errors.SymbolicValueError: Unsupported: ONNX export of convolution for kernel of unknown shape.  [Caused by the value '1113 defined in (%1113 : Float(*, *, *, *, strides=[73728, 73728, 288, 1], requires_grad=1, device=cuda:0) = onnx::Reshap
```

### Dataset

N/A

### Model

N/A

### Steps to reproduce the behavior

Un-skip the `ReverseDistillation` test in `export` and run `tests/integration/model/test_models.py`

### OS information

OS information:
- OS: [e.g. Ubuntu 20.04] 
- Python version: [e.g. 3.8.10]: 3.10
- Anomalib version: [e.g. 0.3.6]: v1 Branch
- PyTorch version: [e.g. 1.9.0]
- CUDA/cuDNN version: [e.g. 11.1]
- GPU models and configuration: [e.g. 2x GeForce RTX 3090]
- Any other relevant information: [e.g. I'm using a custom dataset]

### Expected behavior

Should export successfully.

### Screenshots

_No response_

### Pip/GitHub

GitHub

### What version/branch did you use?

v1

### Configuration YAML

```yaml
NA
```

### Logs

```shell
NA Needs further investigation
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by adrianboguszewski**
## 📝 Description

- Replace kornia gaussian filter which fails when exporting to ONNX with a custom one
- 🛠️ Fixes #1513 

## ✨ Changes

Select what type of change your PR is:

- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/blob/main/docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: src/anomalib/models/image/reverse_distillation/anomaly_map.py
[the remark this answers is not in the document]
Comment (adrianboguszewski) [reply src/anomalib/models/image/reverse_distillation/anomaly_map.py:1]: Done :)

File: src/anomalib/models/image/reverse_distillation/anomaly_map.py [hunk 1]
```diff
@@ -87,8 +88,7 @@ def forward(self, student_features: list[torch.Tensor], teacher_features: list[t
             elif self.mode == AnomalyMapGenerationMode.ADD:
                 anomaly_map += distance_map
 
-        return gaussian_blur2d(
-            anomaly_map,
+        return GaussianBlur2d(
             kernel_size=(self.kernel_size, self.kernel_size),
             sigma=(self.sigma, self.sigma),
-        )
+        )(anomaly_map.to("cpu"))
```
Comment (samet-akcay) [inline src/anomalib/models/image/reverse_distillation/anomaly_map.py:94]: can you elaborate this needs to be explicitly moved to the cpu? Would it be an idea to check the device based on the input?

Comment (adrianboguszewski) [reply src/anomalib/models/image/reverse_distillation/anomaly_map.py:94]: Yep. That's a better idea

**Review by samet-akcay — APPROVED** [review]
Thanks for addressing this!

**Comment by samet-akcay** [comment]
The tests are failing due to device mismatch somewhere

**Comment by adrianboguszewski** [comment]
Let's see if it helps. I cannot test it locally due to a lack of good GPU

**Pull request merged**